libs/ui/design-system/src/lib/components/ShowMoreText/ShowMoreText.tsx

```diff
@@ -0,0 +1,84 @@
+import classNames from "classnames";
+import { CSSProperties, useCallback, useEffect, useRef, useState } from "react";
+
+import "./ShowMoreText.scss";
+
+const CLASS_NAME = "amp-show-more-text";
+
+export type Props = {
+  gapAfterText?: number;
+  lineClamp?: number;
+  children: React.ReactNode;
+};
+
+export const ShowMoreText = ({
+  children,
+  gapAfterText = 20,
+  lineClamp = 3,
+}: Props) => {
+  const ref = useRef<HTMLDivElement>(null);
+
+  const [open, setOpen] = useState(false);
+  const [showLink, setShowLink] = useState(false);
+  const [textHeight, setTextHeight] = useState<number>(0);
+  const [shortHeight, setShortHeight] = useState<number>(0);
```

Comment:
you can set the line-height to 2em => this will give you 2 lines of text. you can also make dynamic by multiple lines coming from props and add style

Reply:
you can remove shortHeight state